apple/swift-docc
# Minimal viable `merge` command

Bug/issue #, if applicable: rdar://114730477

## Summary

This adds a first version of a `docc merge` command that combines documentation archives for different modules into a "combined" documentation archive. 

Two notable features that are missing in this initial version are:
- Merging the LMDB navigator index
- Landing pages (either authored or synthesized) for the combined archive.

However, those can be added incrementally. The merge command in its current form is enough to already be usable and bring value. For example, it allows us to host our documentation for [SwiftDocC](https://apple.github.io/swift-docc/documentation/swiftdocc) and for [SwiftDocCUtilities](https://apple.github.io/swift-docc/documentation/swiftdoccutilities/) from the same repository without breaking their navigator sidebar.

## Dependencies

None

## Testing

- Build documentation for two or more different targets. 

- Run `docc merge /path/to/First.doccarchive /path/to/Second.doccarchive --output-path /path/to/Something.doccarchive`

- Use a local web server to host "Something.doccarchive". 
  - It should be possible to browse both `localhost:PORT/documentation/first` and `localhost:PORT/documentation/second`
    - Each target displays its own assets (if assets had the same name in both)
    - Each target displays a navigator sidebar

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
